api/v1beta1/proposal_webhook.go

```diff
@@ -62,6 +62,11 @@ func (r *Proposal) Default(ctx context.Context, client client.Client, user authe
 	if r.Spec.EndAt.IsZero() {
 		r.Spec.EndAt = metav1.NewTime(time.Now().Add(time.Hour * 24))
 	}
+	if fed, err := AddEndAtAnnotation(client, *r); err == nil {
+		if e := client.Update(context.TODO(), fed); e != nil {
```

Comment:
In Default(), there is no need to update?

Reply:
The webhook is for proposal, the update is  federation, so it should still be needed.